services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/notifications/ConnectionOfTravelNotification/ConnectionOfTravelNotificationRepositoryImpl.java

```diff
@@ -1,79 +1,86 @@
 package us.dot.its.jpo.ode.api.accessors.notifications.ConnectionOfTravelNotification;
 
-import java.util.Date;
-import java.util.List;
-
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Sort;
 import org.springframework.data.mongodb.core.MongoTemplate;
 import org.springframework.data.mongodb.core.query.Criteria;
 import org.springframework.data.mongodb.core.query.Query;
 import org.springframework.stereotype.Component;
-import org.springframework.data.domain.Sort;
+
+import org.springframework.data.domain.Pageable;
 import us.dot.its.jpo.conflictmonitor.monitor.models.notifications.ConnectionOfTravelNotification;
+import us.dot.its.jpo.ode.api.accessors.IntersectionCriteria;
+import us.dot.its.jpo.ode.api.accessors.PageWrapper;
+import us.dot.its.jpo.ode.api.accessors.PaginatedQueryInterface;
 
 @Component
-public class ConnectionOfTravelNotificationRepositoryImpl implements ConnectionOfTravelNotificationRepository {
+public class ConnectionOfTravelNotificationRepositoryImpl
+        implements ConnectionOfTravelNotificationRepository, PageWrapper, PaginatedQueryInterface {
 
     private final MongoTemplate mongoTemplate;
 
-    @Value("${maximumResponseSize}")
-    int maximumResponseSize;
-
     private final String collectionName = "CmConnectionOfTravelNotification";
+    private final String DATE_FIELD = "notificationGeneratedAt";
+    private final String INTERSECTION_ID_FIELD = "IntersectionID";
 
     @Autowired
     public ConnectionOfTravelNotificationRepositoryImpl(MongoTemplate mongoTemplate) {
         this.mongoTemplate = mongoTemplate;
     }
 
-    public Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest) {
-        Query query = new Query();
-
-        if (intersectionID != null) {
-            query.addCriteria(Criteria.where("intersectionID").is(intersectionID));
-        }
-
-        Date startTimeDate = new Date(0);
-        Date endTimeDate = new Date();
-
-        if (startTime != null) {
-            startTimeDate = new Date(startTime);
-        }
-        if (endTime != null) {
-            endTimeDate = new Date(endTime);
-        }
-
-        query.addCriteria(Criteria.where("notificationGeneratedAt").gte(startTimeDate).lte(endTimeDate));
-
-        if (latest) {
-            query.with(Sort.by(Sort.Direction.DESC, "notificationGeneratedAt"));
-            query.limit(1);
-        } else {
-            query.limit(maximumResponseSize);
-        }
-        return query;
+    public long getQueryResultCount(
```

Comment:
chore(blocking): since this doesn't return a query anymore, it makes more sense to rename the method to something like the following. It's up to you what to name it, but `query` doesn' really make sense in the name anymore
```suggestion
    public long count(
```

Reply:
updated